lint job is failing

The lint check is failing. An example can be seen [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/4807121084?check_suite_focus=true#step:8:439)

The error message is the following:
```
golangci-lint run --allow-parallel-runners
level=error msg="Running error: context loading failed: failed to load packages: timed out to load packages: context deadline exceeded"
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
make[2]: *** [../../Makefile.Common:95: lint] Error 4
```

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -75,7 +75,7 @@ jobs:
           path: /home/runner/.cache/go-build
           key: go-build-lint-${{ runner.os }}-${{ hashFiles('**/go.mod', '**/go.sum') }}
       - name: Lint
-        run: make -j4 golint
+        run: make -j2 golint
       - name: Checks 
         run: make -j4 checkdoc porto
       - name: Codegem

